libraries/AP_HAL/AP_HAL_Boards.h

```diff
@@ -382,6 +406,9 @@
 #define HAL_BARO_DEFAULT          HAL_BARO_MS5611_SPI
 #define HAL_BARO_MS5611_NAME "ms5611"
 #define HAL_COMPASS_DEFAULT       HAL_COMPASS_AK8963_MPU9250
+/* HMC5843 can be an external compass */
+#define HAL_COMPASS_HMC5843_I2C_BUS 1
+#define HAL_COMPASS_HMC5843_I2C_ADDR 0x1E
```

Comment:
I just don't want to invest time on this header... still waiting a good soul to add a reasonable config implementation that will get away with it. Meanwhile things get added in the easiest possible way.

I'll eventually get to it and add it, but first I need to finish the pending work I'm doing.